jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/mapper/JPAMapper.java

```diff
@@ -81,23 +87,53 @@ public Long findMapping(Context context, String ownerId) {
         	}
         }
     }
-    
+
+    @Override
+    public Long findLogMapping(Context<?> context, String ownerId) {
+        EntityManagerInfo info = getEntityManager(context);
+        EntityManager em = info.getEntityManager();
+        try {
+            if (context.getContextId() == null) {
+                return null;
+            }
+
+            Query findQuery = em.createNamedQuery("FindContextMapingByAuditContextId", ContextMappingInfo.class)
+                                .setParameter("contextId", context.getContextId().toString())
+                                .setParameter("ownerId", ownerId);
+
+            ContextMappingInfo contextMappingInfo = (ContextMappingInfo) findQuery.getSingleResult();
+            if (contextMappingInfo != null) {
+                return contextMappingInfo.getKsessionId();
+            }
+            return null;
+        } catch (NoResultException e) {
+            return null;
+        } catch (NonUniqueResultException e) {
+            return null;
+        } finally {
+            if (!info.isShared()) {
+                em.close();
+            }
+        }
+    }
 
     @Override
     public void removeMapping(Context context, String ownerId) {
     	EntityManagerInfo info = getEntityManager(context);
     	EntityManager em = info.getEntityManager();
         
-        ContextMappingInfo contextMapping = findContextByContextId(resolveContext(context, em), ownerId, em);
-        if (contextMapping != null) {
-            em.remove(contextMapping);
+        ContextMappingInfo contextMappingInfo = findContextByContextId(resolveContext(context, em), ownerId, em);
+        logger.debug("Removing {}", contextMappingInfo);
+        if (contextMappingInfo != null) {
+            em.remove(contextMappingInfo);
         }
         if (!info.isShared()) {
     		em.close();
     	}
     }
     
-    protected Context resolveContext(Context orig, EntityManager em) {
+    @SuppressWarnings("unchecked")
+    protected <T> Context<T> resolveContext(Context orig, EntityManager em) {
```

Comment:
avoids cast in caller.

Reply:
I tried this locally. If you leave this returning just `Context` and then in the `saveMapping` method just write:

```
Context resolvedContext = resolveContext(context, em);
em.persist(new ContextMappingInfo(resolvedContext.getContextId().toString(),ksessionId, ownerId));
```
It will work without any type parameters.